packages/components/src/internal/util/messaging.tsx

```diff
@@ -168,6 +168,24 @@ export function resolveErrorMessage(
             const fieldname =
                 startIndex > -1 ? 'for field ' + lcMessage.substring(startIndex + 13, lcMessage.indexOf(')=(')) : '';
             return `Unable to create a unique constraint ${fieldname} because duplicate values already exists in the data.`;
+        } else if (
+            errorMsg.indexOf('Invalid SampleSet name.') >= 0 ||
+            errorMsg.indexOf('Invalid DataClass name.') >= 0
```

Comment:
Seems like it might be cleaner to just have two if's, one for samples and one for data classes.

Also, I think for assays we may also want to replace "Domain" with "Assay Design". "Domain" isn't really a noun we use in the application otherwise.